lib/ext/ts.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+var fs = require('fs');
+
+var typescript;
+try {
+	typescript = require('typescript');
+} catch (err) {
+	console.log('AVA error: `--extensions=ts` require TypeScript to be installed.');
+	throw err;
+}
+
+var typescriptOptions = {
+	module: typescript.ModuleKind.CommonJS,
+	target: typescript.ScriptTarget.ES6,
+	sourceMap: true,
+	inlineSources: true,
+	inlineSourceMap: true
+};
+
+/**
+ *
+ * Ts-Node version
+ *
+ */
+function tsnodeRegister() {
+	try {
+		require('ts-node').register(typescriptOptions);
+	} catch (err) {
+		console.log('AVA error: `--extensions=ts` require ts-node to be installed.');
+		throw err;
+	}
+}
+
+function tsnodeTranspile(code, fileName) {
+	try {
+		const tsService = require('ts-node').register(typescriptOptions);
+		const result = tsService().compile(code, fileName);
+		return result;
+	} catch (err) {
+		console.log('AVA error: `--extensions=ts` require ts-node to be installed.');
+		throw err;
+	}
+}
+
+/**
+ *
+ * Typescript version
+ *
+ */
+function typescriptRegister() {
+	require.extensions['.ts'] = function (module, fileName) {
+		var code = fs.readFileSync(fileName, 'utf8');
+		var result = typescript.transpileModule(code, {
+			fileName,
+			compilerOptions: typescriptOptions,
+			reportDiagnostics: true
+		});
+		module._compile(result.outputText, fileName);
+	};
+
+	require('source-map-support').install({
+		hookRequire: true
+	});
+}
```

Comment:
Looks like `ts-node` is closer to `babel-node` than `babel-core` (and `ts-node/register` is equivalent to `babel-core/register`). IMO this module should just use `typescript` to transpile the code, and no other functionality is required.

For consistency we should not automatically transpile source files, at least until we have a better answer for the `.js` use case.